### PR TITLE
Expose structured configuration

### LINKED ISSUE #132: Expose structured configuration for InstaScale controller
## WHY

Currently, InstaScale relies on CLI options for configuration. This prevents the InstaScale controller from being reused in other contexts, as described in project-codeflare/adr#9.

## WHAT

The InstaScale controller should be passed a structured object for its configuration, instead of CLI options.

## HOW

As described in project-codeflare/adr#9.

### PR DESCRIPTION
**Pull request by astefanutti**
# Issue link

Closes #132 

# What changes have been made

* Expose a structured configuration API
* Adapt `main.go` to use the structured configuration 

# Verification steps

Start InstaScale, make sure it works as expected.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -71,7 +78,47 @@ func main() {
...
+	}
+
+	// Source InstaScale ConfigMap
+	if InstaScaleConfigMap, err := kubeClient.CoreV1().ConfigMaps(configsNamespace).Get(ctx, "instascale-config", metav1.GetOptions{}); err != nil {
+		setupLog.Error(err, "Error reading 'instascale-config' ConfigMap")
+		os.Exit(1)
+	} else if maxScaleoutAllowed := InstaScaleConfigMap.Data["maxScaleoutAllowed"]; maxScaleoutAllowed != "" {
+		if max, err := strconv.Atoi(maxScaleoutAllowed); err != nil {
+			setupLog.Error(err, "Error converting 'maxScaleoutAllowed' to integer")
+			os.Exit(1)
```
Comment (anishasthana) [inline main.go:102]: Should instaScale completely error out if the configmap is invalid? Or should it continue on with a default of 3.

Comment (anishasthana) [reply main.go:102]: I guess in a way -- this configmap is something that would be set up by the ODH/RHODS at a later date when the integration is tighter.

**Comment by VanillaSpoon** [comment]
Hey, the `pkg/config` directory is not being copied into the Docker image during the build process causing the build command to fail.

Just a quick change to the Dockerfile:

```diff
# Copy the go source
COPY main.go main.go
COPY controllers/ controllers/
+ COPY pkg/ pkg/
```

**Comment by astefanutti** [comment]
@VanillaSpoon great catch, thanks! Fixed in 0c0fa548c3a8fd25e7ae1fb0e4d6a5ff508dab4c.

**Review by anishasthana — APPROVED** [review]

**Pull request merged**